lib/proxy/log_buffer/decorator.rb

```diff
@@ -1,32 +1,53 @@
 require 'proxy/log_buffer/decorator'
 require 'proxy/log_buffer/buffer'
+require 'thread'
 
 module Proxy::LogBuffer
   class Decorator
+    def self.instance
+      @@instance ||= new(::Proxy::LoggerFactory.logger)
+    end
+
     def initialize(logger, buffer = Proxy::LogBuffer::Buffer.instance)
       @logger = logger
       @buffer = buffer
+      @mutex = Mutex.new
+      @log_rolled = false
+    end
+
+    # due to synchronization can't re-open the log from the signal trap
+    def log_rolled
```

Comment:
Nitpick, this method name is a little unclear as it implies it's been done - perhaps `roll_log`?